diff --git a/src/gasket_interrupt.c b/src/gasket_interrupt.c
index 531d94e..6c62e26 100644
--- a/src/gasket_interrupt.c
+++ b/src/gasket_interrupt.c
@@ -158,7 +158,11 @@ gasket_handle_interrupt(struct gasket_interrupt_data *interrupt_data,
 	read_lock(&interrupt_data->eventfd_ctx_lock);
 	ctx = interrupt_data->eventfd_ctxs[interrupt_index];
 	if (ctx)
+	#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,8,0)
+		eventfd_signal(ctx);
+	#else
 		eventfd_signal(ctx, 1);
+	#endif
 	read_unlock(&interrupt_data->eventfd_ctx_lock);
 
 	++(interrupt_data->interrupt_counts[interrupt_index]);
